src/java.base/share/classes/java/net/Socket.java

```diff
@@ -454,6 +450,10 @@ private Socket(SocketAddress address, SocketAddress localAddr, boolean stream)
         throws IOException
     {
         Objects.requireNonNull(address);
+        if (!stream) {
+            throw new IllegalArgumentException(
+                    "Socket constructor does not support creation of datagram socket");
```

Comment:
```suggestion
                    "Socket constructor does not support creation of datagram sockets");
```

Reply:
Thank you Michael, I updated the PR with all 3 suggestions. The CSR text too has been updated with this minor update.